langchain/text_splitter.py

```diff
@@ -1056,3 +1057,97 @@ def __init__(self, **kwargs: Any) -> None:
         """Initialize a LatexTextSplitter."""
         separators = self.get_separators_for_language(Language.LATEX)
         super().__init__(separators=separators, **kwargs)
+
+
+class GrobidSplitter:
```

Comment:
this follows a completely different API than the text splitter

its fine to not inherit from the base text splitter, but this SHOULD inherit from BaseDocumentTransformer and expose relevant methods there

Reply:
right. there is some confusion on where it should live - 
1/ it was previously in `parsers`, which all inherit from `BaseBlobParser`. but the loader is not a blob loader.
2/ so, we can keep it a `parser` and adapt the loader to yield blobs*.
*i had a quick look at this; we need to pack the file path and resulting XML into a loader.
or
3/ as mentioned, a `text_splitter` may not make sense due to tight coupling to Grobid loader (XML).
4/ but, it could live there and simply inherit from `BaseDocumentTransformer` as mentioned.